weld/src/main/java/org/jboss/as/weld/WeldExtension.java

```diff
@@ -111,7 +112,7 @@ protected boolean rejectAttribute(PathAddress address, String attributeName, Mod
                         //This will not get called if it was discarded, so reject if it is undefined (default==false) or if defined and != 'true'
                         return !attributeValue.isDefined() || !attributeValue.asString().equals("true");
                     }
-                }, WeldResourceDefinition.NON_PORTABLE_MODE_ATTRIBUTE, WeldResourceDefinition.REQUIRE_BEAN_DESCRIPTOR_ATTRIBUTE)
+                }, WeldResourceDefinition.NON_PORTABLE_MODE_ATTRIBUTE, WeldResourceDefinition.REQUIRE_BEAN_DESCRIPTOR_ATTRIBUTE, WeldResourceDefinition.DEVELOPMENT_MODE_ATTRIBUTE)
```

Comment:
I presume that the dev-mode attribute is actually false in the old version, and default is currently false?
In which case this has a different transformer behaviour from the other attributes (those are explained in the comment on line 98). So it needs to be discarded if undefined, and rejected if defined and equals true.